src/pages/category-men/CategoryMen.js

```diff
@@ -0,0 +1,140 @@
+import {useEffect , useState } from "react"
+import axios from "axios"
+import productImg from "../../assets/images/product-image.svg"
+import "../product-listing/ProductListing.css"
+import { Link } from "react-router-dom"
+
+const CategoryMen = () => {
+    const [categoryMenProducts , setCategoryMenProducts ] = useState([])
+    useEffect(() => {
+        axios.get("/api/products").then((response) => {
+            const result = response.data.products
+            setCategoryMenProducts(result.filter(product => product.categoryMen))
+        })
+      }, []);
+    return <div>
+        <header className="header-nav flex">
+        <ul className="navigation navbar-left flex">
+            <li> <Link className="nav-pill" to = "/home">Home</Link> </li>
+            <li><a href="/Product-listing-page/products.html" className="nav-pill">Products</a></li>
+        </ul>
+
+        <input className="search-bar" type="text" placeholder="Search" />
+
+        <ul className="navigation navbar-right flex">
+            <li>
+                <div className="badge-on-cart wishlist-icon">
+                    <a href="/Wishlist/wishlist.html" className="nav-pill"><i className="material-icons"> favorite_border</i></a>
+                    <p className="icon-badge aligned-icon">5</p>
+                </div>   
+            </li>
+
+            <li>
+                <a href="/Authentication/login.html" className="nav-pill account-icon"><i class="material-icons"> account_circle</i></a>
+            </li>
+
+            <li>
+                <div className="badge-on-cart cart-icon">
+                    <a href="/Cart/cart.html" className="nav-pill"><i className="material-icons"> add_shopping_cart</i></a>
+                    <p className="icon-badge aligned-icon">1</p>
+                </div>
+                
+            </li>
+
+        </ul>
+    </header>
+
+    <section className="page-container flex-page">
+        <aside className="filters">
+            <div className="filter flex ">
+                <p className="filter-heading margin-left margin-bottom margin-top">Filters</p>
+                <a className="clear-link margin-right margin-bottom margin-top" href="#">Clear</a>
+            </div>
+            <div className="filter">
+                <p className="filter-heading-text margin-left">Category</p>
+                <div className="flex">
+                    <input className="margin-left margin-top margin-bottom" id="category-checkbox1" name = "category"
+                        type="checkbox" checked={showCategoryMen}
+                        onChange={() => dispatch({ type: "CATEGORY_MEN" })}/>
```

Comment:
hey where is this dispatch declared that u are calling it here. cause u have used a useState and dispatch is usually used with a useReducer . so didnt quite from where is this dispatch getting called